lib/archethic/oracle_chain/services/uco_price.ex

```diff
@@ -21,30 +23,34 @@ defmodule Archethic.OracleChain.Services.UCOPrice do
     {:ok, fetching_tasks_supervisor} = Task.Supervisor.start_link()
     ## retrieve prices from configured providers and filter results marked as errors
     prices =
-      Task.Supervisor.async_stream_nolink(
-        fetching_tasks_supervisor,
-        providers(),
-        fn provider ->
-          case provider.fetch(@pairs) do
-            {:ok, _prices} = result ->
-              result
-
-            {:error, reason} ->
-              provider_name = provider |> to_string() |> String.split(".") |> List.last()
-
-              Logger.warning(
-                "Service UCOPrice cannot fetch values from " <>
-                  "provider: #{inspect(provider_name)} with reason : #{inspect(reason)}."
-              )
-
-              {:error, provider}
-          end
-        end,
-        on_timeout: :kill_task
-      )
-      |> Stream.filter(&match?({:ok, {:ok, _}}, &1))
-      |> Stream.map(fn
-        {_, {_, result = %{}}} ->
+      Enum.map(providers(), fn provider ->
+        case HydratingCache.get(
+               :"Elixir.Archethic.Utils.HydratingCache.uco_service",
```

Comment:
Why not using direct atom ?

Reply:
:) Right, this is definitely something I need to refine